Add faq card type to default cards

ljewalsh opened the pull request:
The new card has three fields:
- name (for the FAQ question)
- answer (markdown so we can add complex styling to our answers)
- topic (so we can group FAQ questions and answers together)

For now topic is a simple string. I was not sure whether the grouping could occur yet so I left everything as a simple view

There also didn't seem to be any tests for the other cards. I'd be happy to add some simple tests (e.g. did this card load when the server starts?). Let me know if this is something you'd like

**Please remember to write tests for your changes**. We aim to automatically
deploy `master` to production, and we can't safely do this without a solid
test suite!

Commit:
core: Warn on slugs without versions

Change-type: major
Signed-off-by: Juan Cruz Viotti <juan@balena.io>

LucianBuzzo committed:
doc: Add documentation for data model, permissions and API queries

Change-type: patch
Signed-off-by: Lucian Buzzo <lucian.buzzo@gmail.com>

ffissore committed:
Use graphile-worker instead of our own implementation of a db based
queue

Change-type: minor
Signed-off-by: Federico Fissore <federico@balena.io>

ljewalsh committed:
Add faq card type to default cards

Change-type: minor
Signed-off-by: Lucy-Jane Walsh <ljewalsh@gmail.com>

ljewalsh committed:
Upgrade deprecated Octokit headers and properties

Change-type: patch
Signed-off-by: Lucy-Jane Walsh <ljewalsh@gmail.com>

AlidaOdendaal committed:
add-new-type: Fix minor typos

Change-type: patch
Signed-off-by: AlidaOdendaal <alida@balena.io>

AlidaOdendaal committed:
default-cards: Add pipeline card type with associated view

Change-type: minor
Signed-off-by: AlidaOdendaal <alida@balena.io>
Connects-to: #2854

ffissore committed:
Don't use `process.env` as UI env vars container, because it
interferes with some third party modules using it: the result is a
different `vendors` bundle for each release of the UI, even if no
changes involved third party libraries

Connects-to: #3134
Change-type: patch
Signed-off-by: Federico Fissore <federico@balena.io>

ljewalsh committed:
Merge branch 'master' into faq-card

ljewalsh committed:
Merge branch 'master' into faq-card

balena-ci committed:
v25.1.1

grahammcculloch reviewed (changes requested):
@ljewalsh I think you'll need/want to squash the 3 commits into one (to appease the VersionBot Gods). I just found [this excellent guide](https://www.internalpointers.com/post/squash-commits-into-one-git) that walks you through doing it in case (like me) you aren't super familiar with the process.

ljewalsh committed:
Merge branch 'master' into faq-card

balena-ci committed:
v26.0.0

grahammcculloch committed:
ui: Fixed 'Add x' button on Kanban view

Clicking this button now correctly opens the form for creating a new card.

Change-type: patch
Signed-off-by: Graham McCulloch <graham@balena.io>

joshbwlng committed:
Add document about running tests

Change-type: patch
Signed-off-by: Josh Bowling <josh@balena.io>

balena-ci committed:
v26.0.1

grahammcculloch committed:
dev: Update repository code owners

Added grahammcculloch as a code owner

Change-type: patch
Signed-off-by: Graham McCulloch <graham@balena.io>

balena-ci committed:
v26.0.2

balena-ci committed:
v26.1.0

balena-ci committed:
v26.1.1

balena-ci committed:
v26.1.2

ffissore committed:
Remove duplicate "Outreach Mirror" test executions

Change-type: patch
Signed-off-by: Federico Fissore <federico@balena.io>

balena-ci committed:
v26.2.0

ljewalsh committed:
Merge branch 'master' into faq-card

joshbwlng committed:
Use postgres host env var in more places

Change-type: patch
Signed-off-by: Josh Bowling <josh@balena.io>

balena-ci committed:
v26.2.1

ffissore committed:
Fix bold mistake introduced in e03c806f5b2204f1c27b43899b34953f53e10845

We are not currently logging unless levels matche, but that's NOT how
log levels work: if log level is 'debug' and we log an exception, it
must be logged

Change-type: patch
Signed-off-by: Federico Fissore <federico@balena.io>

balena-ci committed:
v26.2.2

balena-ci committed:
v26.2.3

StefKors committed:
Remove webpack @jellyfish alias from ui-components

Change-type: patch
Signed-off-by: Stef Kors <stef@balena.io>

StefKors committed:
Update webpack.config to remove the @jellyfish alias

Change-type: patch
Signed-off-by: Stef Kors <stef@balena.io>

StefKors committed:
Allow window to be undefined

Change-type: patch
Signed-off-by: Stef Kors <stef@balena.io>

StefKors committed:
Update unit tests to support future sinon versions

Change-type: patch
Signed-off-by: Stef Kors <stef@balena.io>

StefKors committed:
Fix CardChatSummery test

Change-type: patch
Signed-off-by: Stef Kors <stef@balena.io>

balena-ci committed:
v26.2.4

balena-ci committed:
v26.2.5

joshbwlng committed:
Update references to GitHub test repo to new location

Change-type: patch
Signed-off-by: Josh Bowling <josh@balena.io>

balena-ci committed:
v26.2.6

balena-ci committed:
v26.2.7

balena-ci committed:
v26.2.8

ljewalsh committed:
Use random-word package to diversify data

Change-type: patch
Signed-off-by: Lucy-Jane Walsh <ljewalsh@gmail.com>

balena-ci committed:
v26.2.9

balena-ci committed:
v26.2.10

ljewalsh committed:
Merge branch 'master' into faq-card

ljewalsh committed:
Merge branch 'faq-card' of https://github.com/balena-io/jellyfish into faq-card
